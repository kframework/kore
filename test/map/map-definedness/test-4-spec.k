module VERIFICATION
  imports TEST
endmodule

module TEST-4-SPEC
  imports VERIFICATION

  rule
    <k> add(f(X), f(3)) => end </k>
    <a> 2 |-> 3 M => ?_ </a>

endmodule
