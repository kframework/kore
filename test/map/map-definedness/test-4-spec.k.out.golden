  #Ceil ( f ( 3 ) )
#And
  #Ceil ( f ( X ) )
#And
  #Not ( #Not ( {
      2
    #Equals
      f ( X )
    } )
  #And
    {
      f ( X ) in_keys ( M )
    #Equals
      false
    }
  #And
    {
      false
    #Equals
      2 in_keys ( M )
    } )
#And
  <T>
    <k>
      added
    </k>
    <a>
      2 |-> 3
      M
      f ( X ) |-> f ( 3 )
    </a>
  </T>
#And
  {
    2 in_keys ( M )
  #Equals
    false
  }
